std-bits/table/src/main/java/org/enso/table/data/column/storage/StringStorage.java

```diff
@@ -46,6 +54,29 @@ public TextType getType() {
     return type;
   }
 
+  /**
+   * Counts the number of cells in the columns with whitespace. If the calculation fails then it
+   * returns null.
+   *
+   * @return the number of cells with whitespace
+   */
+  public Long countUntrimmed() throws InterruptedException {
+    if (countUntrimmed.isCancelled()) {
+      // Need to recompute the value, as was cancelled.
+      countUntrimmed =
+          CompletableFuture.supplyAsync(
+              () ->
+                  CountUntrimmed.compute(
+                      this, CountUntrimmed.DEFAULT_SAMPLE_SIZE, Context.getCurrent()));
```

Comment:
Good catch. But at this point I don't think computing on background thread is correct. We are blocking the main thread waiting for this to compute, but the background thread may actually be stalled with many other queued computations.

It will be much more efficient to just compute on the main thread then.

> Also: if you are computing this in the background, then you need to pass `null` as `Context`, as we remembered that getting Truffle Context on a background thread will not end well.

```suggestion
      // Need to recompute the value, as was cancelled.
      countUntrimmed =
          CompletableFuture.completedFuture(
          		CountUntrimmed.compute(this, CountUntrimmed.DEFAULT_SAMPLE_SIZE, Context.getCurrent())
          );
```

IMO it's much better to compute this on foreground thread if the background tasks was cancelled. It's simpler and possibly faster. And then we can actually pass the `Context` because we are on main thread.